src/widgets/PieChart.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2018
+ */
+
+import React, { useState, useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import { VictoryLabel, VictoryPie } from 'victory-native';
+import { APP_FONT_FAMILY, GREY } from '../globalStyles';
+
+/**
+ * A charting widget for displaying Pie chart report.
+ *
+ * @prop  {string}       id      The report ID.
+ * @prop  {string}       title   The title of the report.
+ * @prop  {string}       type    The type of chart to use to display the report,
+ *                               options are BarChart, LineChart and PieChart.
+ * @prop  {data}         array   An array of {x, y} datapoints to plot.
+ * @prop  {width}        number  The width of the parent container.
+ * @prop  {height}       number  The height of the parent continer.
+ */
+
+export const ReportChart = ({ report: initialReport }, props) => {
+  const [report, setReport] = useState(initialReport);
```

Comment:
keeping values in state as well as props is generally an anti-pattern.

If you think about it, it doesn't really make sense to have the value passed to your component also held in state.. you're just duplicating the data for no real reason!

Reply:
Corrected here 0f3f2a7
Not totally sure about this one:
`const [state, setState] = useState({ height: null, width: null });`

Reply:
Its name can cause confusions, what do you think? Maybe `heightAndWidth` or `layoutSizes`, etc can represent them better than just `state`?